### PR TITLE
Enabling forking in gom files

### PR DESCRIPTION
**Pull request by SimonRichardson**
The idea is to enable a repo represent another repo. In the vien forking
allows a github repo to use a possible downstream fork.
- Make sure we move after go get

---

Example:

```
gom 'github.com/stripe/stripe-go', :fork => 'github.com/dicefm/stripe-go', :branch => 'dynamic-last-four'
```

### CODE REVIEW

File: Gomfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+gom 'github.com/stripe/stripe-go', :fork => 'github.com/dicefm/stripe-go', :branch => 'dynamic-last-four'
```
Comment (iceydee) [inline Gomfile]: @SimonRichardson do you want that as part of the GOM change?

Comment (SimonRichardson) [inline Gomfile]: Probably not.

Comment (SimonRichardson) [inline Gomfile]: Also need to work out how we bring our changes into the upstream, we've diverged a bit atm.

**Comment by iceydee** [comment]
:+1:

**Pull request merged**